libs/utils/android/workloads/youtube.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+
+from subprocess import Popen, PIPE
+from android import Screen, Workload
+from time import sleep
+
+import logging
+
+YOUTUBE_CMD = 'shell dumpsys gfxinfo com.google.android.youtube > {}'
+
+class YouTube(Workload):
+    """
+    Android YouTube workload
+    """
+
+    # Package required by this workload
+    package = 'com.google.android.youtube'
+
+    # Setup logger
+    logger = logging.getLogger('YouTube')
+    logger.setLevel(logging.INFO)
+
+
+    def __init__(self, test_env):
+        super(YouTube, self).__init__(test_env)
+        logging.debug('%14s - Workload created', 'YouTube')
+
+    def run(self, exp_dir, video_url, video_duration_s, collect=''):
+
+        # Initialize energy meter results
+        nrg_data, nrg_file = None, None
+
+        # Unlock device screen (assume no password required)
+        self.target.execute('input keyevent 82')
+        # Press Back button to be sure we run the video from the start
+        self.target.execute('input keyevent KEYCODE_BACK')
+
+        # Force screen in LANDSCAPE mode
+        Screen.set_orientation(self.target, portrait=False)
+
+        # Start YouTube video on the target device
+        youtube_cmd = 'am start -a android.intent.action.VIEW "{}"'\
+                      .format(video_url)
+        logging.info(youtube_cmd)
+        self.target.execute(youtube_cmd)
+        # Allow the activity to start
+        sleep(3)
```

Comment:
We may have problems with that in the future.  Maybe not for now, but we should consider making this a parameter so that the user can specify a longer start time if they know how long it takes it to start in their device.